webpack/assets/javascripts/react_app/redux/API/APIRequest.js

```diff
@@ -46,6 +52,13 @@ export const apiRequest = async (
       response: response.data,
     });
 
+    updateData &&
```

Comment:
Can you also check if the state for the API key exists?
e.g.:
```suggestion
   const prevResponse = selectAPIResponse(prevState, key);
   if (updateData && !isEmpty(prevResponse)) dispatch....
```

another option would be to display show an error in the console.

Reply:
The state can be empty, consumers can modify just the current response's data,
I don't think that raising an error is needed here, if a consumer will use an unexisting state it would be just empty object (and trying to access undefined keys will end with an error).

```js
  post({
    key: MY_SPECIAL_KEY,
    url,
    updateData: (prevResponseState, currentResponse) => ({
      results: currentResponse.map(someCallback),
    }),
```

Mostly in this case it would be better to change the return format in the server, though we don't change the API often.

Reply:
rethinking about it, the namespace will always be there as it's being dispatched after the REQUEST stage, 
so no concern here